pkg/zonemgr/zone_file.go

```diff
@@ -2,180 +2,151 @@ package zonemgr
 
 import (
 	"fmt"
-	v1 "kubevirt.io/api/core/v1"
+	"k8s.io/utils/net"
+	"sort"
 	"strconv"
+
+	v1 "kubevirt.io/api/core/v1"
 )
 
 const (
 	refresh = "3600"    // 1 hour (seconds) - how long a nameserver should wait prior to checking for a Serial Number increase within the primary zone file
 	retry   = "3600"    // 1 hour (seconds) - how long a nameserver should wait prior to retrying to update a zone after a failed attempt.
 	expire  = "1209600" // 2 weeks (seconds) - how long a nameserver should wait prior to considering data from a secondary zone invalid and stop answering queries for that zone
 	ttl     = "3600"    // 1 hour (seconds) - the duration that the record may be cached by any resolver
+
+	domainDefault     = "vm"
+	nameServerDefault = "ns"
+	adminEmailDefault = "email"
 )
 
 type ZoneFile struct {
-	soaSerial          int
-	clusterName        string
-	userAdminEmail     string
-	userNameserverName string
-	userNameserverIP   string
-	userSubdomain      string
-
-	domainName string
+	soaSerial      int
+	adminEmail     string
+	nameServerName string
+	nameServerIP   string
+	domain         string
+
 	headerPref string
 	headerSuf  string
 
 	header   string
 	aRecords string
 	content  string
 
-	vmiRecordsMap map[string]map[string]ARecord // key=vmi_name+namespace, value=map[key=iface_name, value=ip]
-	//ifacesMap map[string]string
+	vmiRecordsMap map[string][]string // key=vmi_name+namespace, value=ARecords array
 }
 
-func NewZoneFile(clusterName string, userAdminEmail string, userNameserverName string,
-	userNameserverIP string, userSubdomain string) *ZoneFile {
+func NewZoneFile(nameServerIP string, domain string) *ZoneFile {
 	return &ZoneFile{
-		clusterName:        clusterName,
-		userAdminEmail:     userAdminEmail,
-		userNameserverName: userNameserverName,
-		userNameserverIP:   userNameserverIP,
-		userSubdomain:      userSubdomain,
+		nameServerIP: nameServerIP,
+		domain:       domain,
 	}
 }
 
 func (zoneFile *ZoneFile) init() {
-	zoneFile.setDefaultValues()
+	zoneFile.initCustomFields()
 	//before SOA serial
 	zoneFile.generateHeaderPrefix()
 	//after SOA serial
 	zoneFile.generateHeaderSuffix()
 	zoneFile.soaSerial = 0 // TODO make default RANDOM: time.Now().UnixNano()
```

Comment:
please remove the comment,
also no need random

please add logic that increment the SOA serial upon each write
(the robustness of container restart can be done on a follow PR)

Reply:
done